### PR TITLE
Rely on libc for correct integer types in os/unix/net/ancillary.rs.

### PR DESCRIPTION
**Pull request by de-vri-es**
This PR is a small maintainability improvement. It simplifies `unix/net/ancillary.rs` in `std` by removing the `cfg_ifs` for casting to the correct integer type, and just rely on libc to define the struct correctly.

### CODE REVIEW

File: library/std/src/os/unix/net/ancillary.rs [hunk 1]
```diff
@@ -364,25 +326,7 @@ impl<'a> AncillaryData<'a> {
 
     fn try_from_cmsghdr(cmsg: &'a libc::cmsghdr) -> Result<Self, AncillaryError> {
         unsafe {
-            cfg_if::cfg_if! {
-                if #[cfg(any(
-                        target_os = "android",
-                        all(target_os = "linux", target_env = "gnu"),
-                        all(target_os = "linux", target_env = "uclibc"),
```
Comment (m-ou-se) [inline library/std/src/os/unix/net/ancillary.rs:371]: This `cfg` is different than the others.

**Comment by m-ou-se** [comment]
Nice! The fewer of these `cfg` blocks we have, the better. :)

File: library/std/src/os/unix/net/ancillary.rs (hunk 1 above)
Comment (de-vri-es) [reply library/std/src/os/unix/net/ancillary.rs:371]: After some consideration, I decided to remove the `cfg_if` entirely and rely on libc to have the correct struct definition instead. If something turns out to be incorrect, it only needs to be fixed in one place :)

File: library/std/src/os/unix/net/ancillary.rs [hunk 2]
```diff
@@ -364,28 +306,10 @@ impl<'a> AncillaryData<'a> {
...
-                          ))] {
-                    let cmsg_len_zero = libc::CMSG_LEN(0) as libc::socklen_t;
-                }
-            }
-            let data_len = (*cmsg).cmsg_len - cmsg_len_zero;
+            let cmsg_len_zero = libc::CMSG_LEN(0) as usize;
+            let data_len = (*cmsg).cmsg_len as usize - cmsg_len_zero;
             let data = libc::CMSG_DATA(cmsg).cast();
-            let data = from_raw_parts(data, data_len as usize);
+            let data = from_raw_parts(data, data_len);
```
Comment (de-vri-es) [inline library/std/src/os/unix/net/ancillary.rs:312]: This can just always be a usize, because it's only passed to `slice::from_raw_parts`.

**Comment by m-ou-se** [comment]
Thanks!

@bors r+

**Pull request merged**